gen3_tracker/git/cli.py

```diff
@@ -0,0 +1,856 @@
+                        'urls': _['urls']
+                     } for _ in indexd_records
+                ]
+
+        bucket_ids = {_['did'] for _ in indexd_records}
+
+        uncommitted = pathlib.Path('MANIFEST').glob('**/*.dvc')
+        uncommitted = [str(_) for _ in uncommitted]
+        uncommitted = [str(_) for _ in uncommitted if _ not in committed_files]
+        uncommitted = [_.model_dump(exclude_none=True) for _ in dvc_data(uncommitted)]
+
+        _ = {
+            'bucket': indexd_records,
+            'committed': [_dvc_meta(v, full=True) for k, v in dvc_objects.items() if k not in bucket_ids],
+            'uncommitted': uncommitted
+        }
+
+        if target:
+            # Escape special characters and replace wildcard '*' with '.*' for regex pattern
+            pattern = re.escape(target).replace("\\*", ".*")
+            filtered = {
+                'bucket': filter_dicts(_.get('bucket', []), pattern),
+                'committed': filter_dicts(_.get('committed', []), pattern),
+                'uncommitted': filter_dicts(_.get('uncommitted', []), pattern)
+            }
+            _ = filtered
+
+        if config.output.format == 'json':
+            print(json.dumps(_, indent=2))
+        else:
+            yaml.dump(_, sys.stdout, default_flow_style=False)
+
+    except Exception as e:
+        click.secho(str(e), fg=ERROR_COLOR, file=sys.stderr)
+        if config.debug:
+            raise
+
+
+@cli.command()
+@click.argument('object_id', metavar='<name>')
+@click.pass_obj
+def rm(config: Config, object_id: str):
+    """Remove a single file from the server index, and MANIFEST. Does not alter META.
+    \b
+    <name> is a GUID or a data file name.
+    """
+    try:
+
+        with Halo(text='Searching', spinner='line', placement='right', color='white'):
+            object_id, path = file_name_or_guid(config, object_id)
+
+        with Halo(text='Deleting from server', spinner='line', placement='right', color='white'):
+            auth = gen3_tracker.config.ensure_auth(config=config)
+            index = Gen3Index(auth)
+            result = index.delete_record(object_id)
+        if not result:
+            if not path:
+                path = ''
+            click.secho(f"Failed to delete {object_id} from server. {path}", fg=ERROR_COLOR, file=sys.stderr)
+        else:
+            click.secho(f"Deleted {object_id} from server. {path}", fg=INFO_COLOR, file=sys.stderr)
+
+        with Halo(text='Scanning', spinner='line', placement='right', color='white'):
+            committed_files, dvc_objects = manifest(config.gen3.project_id)
+            dvc_objects = [dvc_object for dvc_object in dvc_objects if dvc_object.object_id == object_id]
+            assert dvc_objects, f"{object_id} not found in MANIFEST."
+            dvc_object = dvc_objects[0]
+            path = pathlib.Path('MANIFEST') / (dvc_object.out.path + ".dvc")
+            assert path.exists(), f"{path} not found"
+            path.unlink()
```

Comment:
See how this is where the deleted files are removed from the Manifest, but how is this information passed to the server?

Reply:
See https://github.com/ACED-IDP/gen3_util/blob/7bc05e8c2d8cfb3f01ad415c9bea068052bd96b9/tests/integration/test_bundle.py